### PR TITLE
Feature/incorporate details

### PR DESCRIPTION
**Pull request by alia-peterson**
## *Purpose:*
*What does this merge do? Check all that apply.*
- [x] Adds a new feature
- [x] Improves an existing feature
- [ ] Fixes a bug
- [ ] Cleans up (logs, tests, etc.)

## *Description: What does this PR do?*
Added a controlled form so that a user can update their profile information 

## *What are the relevant tickets?*
N/a

## *Bugs:*
No known bugs

## Checklist:
- [x] I have performed a self-review of my own code
- [ ] I have used tests to prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my change

### CODE REVIEW

File: code/web/src/modules/user/Profile.js [hunk 1]
```diff
@@ -52,16 +52,42 @@ class Profile extends PureComponent {
...
+    }
   }
 
   // Runs on server only for SSR
   static fetchData({ store }) {
     return store.dispatch(getListByUser())
   }
 
+  // Update controlled form
+  handleChange = (event) => {
```
Comment (elsafluss) [inline code/web/src/modules/user/Profile.js:66]: great controlled form

File: code/web/src/modules/user/Profile.js [hunk 2]
```diff
@@ -104,27 +130,112 @@ class Profile extends PureComponent {
               marginLeft: '2em',
               padding: '2em',
               textAlign: 'center',
-              objectFit: 'contain',
               maxWidth: 250,
-              maxHeight: 250
+              maxHeight: 250,
             }}>
-            <ImageTile height={250} width={250} image={userImage} shadow={level3}/>
+            <ImageTile
```
Comment (elsafluss) [inline code/web/src/modules/user/Profile.js:136]: glad we got this one figured out

File: code/web/src/modules/user/api/actions.js [hunk 3]
```diff
@@ -11,6 +11,7 @@ export const LOGIN_REQUEST = 'AUTH/LOGIN_REQUEST'
 export const LOGIN_RESPONSE = 'AUTH/LOGIN_RESPONSE'
 export const SET_USER = 'AUTH/SET_USER'
 export const LOGOUT = 'AUTH/LOGOUT'
+export const TOGGLE_FORM = 'AUTH/TOGGLE_FORM'
```
Comment (elsafluss) [inline code/web/src/modules/user/api/actions.js:14]: makin our own actions!

File: code/web/src/modules/user/api/actions.js [hunk 4]
```diff
@@ -36,7 +37,7 @@ export function login(userCredentials, isLoading = true) {
     return axios.post(routeApi, query({
       operation: 'userLogin',
       variables: userCredentials,
-      fields: ['user {name, email, role}', 'token']
+      fields: ['user {name, email, role, description, addressLine1, addressLine2, city, state, zipcode, image}', 'token']
```
Comment (elsafluss) [inline code/web/src/modules/user/api/actions.js:40]: gettin that API data!

File: code/web/src/modules/user/api/actions.js [hunk 5]
```diff
@@ -87,6 +88,26 @@ export function register(userDetails) {
   }
 }
 
+// Update user data
+export function updateUser(userDetails) {
+  return dispatch => {
+    return axios.post(routeApi, mutation({
```
Comment (elsafluss) [inline code/web/src/modules/user/api/actions.js:94]: we aren't calling this one yet, but I believe in us

File: code/web/src/modules/user/api/state.js [hunk 6]
```diff
@@ -43,7 +44,13 @@ export default (state = userInitialState, action) => {
         details: null
       }
 
+    case TOGGLE_FORM:
```
Comment (elsafluss) [inline code/web/src/modules/user/api/state.js:47]: makin our own reducer!

File: code/web/src/ui/image/Tile.js [hunk 7]
```diff
@@ -26,14 +26,13 @@ const Tile = (props) => {
 Tile.propTypes = {
   image: PropTypes.string.isRequired,
   style: PropTypes.object,
-  width: PropTypes.number,
-  height: PropTypes.number,
   shadow: PropTypes.string
```
Comment (elsafluss) [inline code/web/src/ui/image/Tile.js:29]: you can't require width and height to be a number and then set its default to a string.

**Review by elsafluss — APPROVED** [review]

File: code/web/src/modules/user/Profile.js [hunk 8]
```diff
@@ -11,6 +11,8 @@ import { H3, H4 } from '../../ui/typography'
 import Button from '../../ui/button'
 import ImageTile from '../../ui/image/Tile'
 import Card from '../../ui/card/Card'
+import Input from '../../ui/input/Input'
+import Textarea from '../../ui/input/Textarea'
```
Comment (mattdeann) [inline code/web/src/modules/user/Profile.js:15]: Can we style this better?

File: code/web/src/modules/user/Profile.js (hunk 1 above)
Comment (mattdeann) [reply code/web/src/modules/user/Profile.js:66]: for real i am stunned.

File: code/web/src/modules/user/Profile.js [hunk 9]
```diff
@@ -52,16 +52,42 @@ class Profile extends PureComponent {
...
+
+  // Submit controlled form data
+  handleSubmit = () => {
+    this.props.toggleForm()
+  }
+
   // Runs on client only
   componentDidMount() {
     this.props.getListByUser()
+    const userImage = `https://media.newyorker.com/photos/5e49bf473399bf0008132231/1:1/w_2539,h_2539,c_limit/Kenseth-CatProfile.jpg`
```
Comment (mattdeann) [inline code/web/src/modules/user/Profile.js:83]: should we hold this somewhere else / as a global var for the file?

File: code/web/src/modules/user/Profile.js [hunk 10]
```diff
@@ -104,27 +130,112 @@ class Profile extends PureComponent {
...
               padding: '2em',
               textAlign: 'center',
-              objectFit: 'contain',
               maxWidth: 250,
-              maxHeight: 250
+              maxHeight: 250,
             }}>
-            <ImageTile height={250} width={250} image={userImage} shadow={level3}/>
+            <ImageTile
+              style={{backgroundRepeat: 'no-repeat', backgroundSize: 'cover', backgroundPosition: 'center'}}
```
Comment (mattdeann) [inline code/web/src/modules/user/Profile.js:137]: ahhhh this makes sense, working with JSX styled components is a bit more work than anticipated!

**Review by mattdeann — APPROVED** [review]
- [x] Changes generate no new warnings or errors
- [x] Branch was tested locally without failures
- [ ] All tests are passing as expected

**Pull request merged**